src/main/java/com/wiley/autotest/services/WebServiceMethodsInvoker.java

```diff
@@ -44,14 +42,22 @@ void invokeMethod(AbstractTest instance, Method method, TestClassContext context
             method.setAccessible(true);
             method.invoke(instance);
         } catch (Throwable e) {
-            String errorMessage = format("Precondition method '%s' failed ", method.getName()) + "\n " + ExceptionUtils.getStackTrace(e);
+            Throwable exceptionForLog;
+            if (e instanceof InvocationTargetException) {
+                exceptionForLog = ((InvocationTargetException) e).getTargetException();
+            } else {
+                exceptionForLog = e;
+            }
+
+            String errorMessage = "Precondition method failed " + exceptionForLog.getMessage();
```

Comment:
@FedVl @estuold do you know how do we define that we failed exactly in a precondition method? method name is just "invokeMethod" so will it be used for any method?

Reply:
invokeMethod used only for call mehtods with our custom @Before... and @After...